### ISSUE TITLE
ConcurrentModificationException from AutoPushFeature.notifyObservers

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#8950]
https://sentry.prod.mozaws.net/operations/fenix-nightly/issues/7424258/

```
ConcurrentModificationException: None
    at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:757)
    at java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:780)
    at mozilla.components.support.base.observer.ObserverRegistry.notifyObservers(ObserverRegistry.kt:2)
    at mozilla.components.feature.push.AutoPushFeature.notifyObservers
    at mozilla.components.feature.push.AutoPushFeature$onMessageReceived$1$1.invokeSuspend(AutoPushFeature.kt:22)
...
(7 additional frame(s) were not displayed)
```

**Comment by AndiAJ** [comment]
Hi @jonalmeida , could you please help us out with a STR? ☺️

**Comment by jonalmeida** [comment]
Sorry @AndiAJ!

### STR

1. Login to a Firefox Account.
2. Send a tab to the device.
3. Swipe the app away from the app switcher.
4. Open the app and immediately send a tab to the device.

### Expected

 - The app shouldn't crash.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on the latest **`Nightly Build #20710605 from 03/11`** using the following devices:
• Google Pixel 3a (Android 9)
• Huawei Mate 20 Lite (Android 8.1.0)
• Samsung Galaxy S7 (Android 7)
• OnePlus A3 (Android 6.0.1)


►Video
![20200311-113750](https://user-images.githubusercontent.com/51314259/76406514-8373c980-6392-11ea-8bff-23b275d2beac.gif)
